client/engine/messageservice/test-messageservice.go

```diff
@@ -80,7 +82,27 @@ func (t TestMessageService) forward(message protocols.Message) {
 // routeOutgoing listens to the messageService's outbox and passes
 // messages to the forwarding function
 func (t TestMessageService) routeOutgoing() {
-	for msg := range t.out {
+	for msg := range t.in {
 		t.forward(msg)
 	}
 }
+
+// ┌──────────┐           in┌───────────┐
+// │          │  ───────────►           │
+// │  Engine  │             │  Message  │
+// │          │          out│  Service  │
+// │    A     │  ◄──────────┤    A      │
+// └──────────┘             └────┬──────┘
+//                               │toPeers[B]
+//                               │
+//                               │
+//                  ┌────────────┘
+//                  │
+//                  │
+//                  │
+// ┌──────────┐     │     in┌───────────┐
+// │          │  ───┼───────►           │
+// │  Engine  │     │       │  Message  │
+// │          │     │    out│  Service  │
+// │    B     │  ◄──┴───────┤    B      │
+// └──────────┘             └───────────┘
```

Comment:
```suggestion

// ┌──────────┐toMsg       in┌───────────┐
// │          │─────────────►|           │
// │  Engine  │              │  Message  │
// │          │fromMsg    out│  Service  │
// │    A     │◄─────────────┤    A      │
// └──────────┘              └────┬──────┘
//                                │toPeers[B]
//                                │
//                                │
//                     ┌──────────┘
//                     │
//                     │
//                     │
// ┌──────────┐toMsg   │   in┌───────────┐
// │          │────────┼────►|           │
// │  Engine  │        │     │  Message  │
// │          │fromMsg │  out│  Service  │
// │    B     │◄───────┴─────┤    B      │
// └──────────┘              └───────────┘
```

Maybe: include the name that each channel has from each perspective